tanjun/clients.py

```diff
@@ -1230,6 +1271,16 @@ def remove_component(self, component: tanjun_abc.Component, /) -> None:
                 self.dispatch_client_callback(ClientCallbackNames.COMPONENT_REMOVED, component)
             )
 
+    # TODO: do we want to assert that component names are unique
+    def remove_component_by_name(self, name: str, /) -> None:
```

Comment:
Could this just be merged with move_component_by_name if we assert that names are unique?